crates/handlers/src/compat/login.rs

```diff
@@ -97,6 +122,9 @@ pub struct RequestBody {
 
     #[serde(default)]
     refresh_token: bool,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    device_id: Option<String>,
```

Comment:
When login we pass a device_id unique to each user so we have a unique session per user coming from our web platform (otherwise each login to the platform would stack as a new session in Element and ask to be checked again which would be a mess).

Since it's actually in [the Matrix specification](https://spec.matrix.org/v1.12/client-server-api/#relationship-between-access-tokens-and-devices), I felt like it was actually a missing piece from the compat create. But yes, maybe it should sit in its own PR.